### PR TITLE
8035395: sun/management/jmxremote/startstop/JMXStartStopTest.java fails intermittently: Port already in use

### PR DESCRIPTION
**Pull request by ktakakuri**
Hi all,

This is a backport of JDK-8035395: sun/management/jmxremote/startstop/JMXStartStopTest.java fails intermittently: Port already in use.
The JMXStartStopTest.java test failures reported in this issue also occur in JDK8.
This backport improves testing to use a dynamically allocated port.

testing: jdk/test/sun/management/jmxremote/startstop/JMXStartStopTest.java

With this fix alone, the test remains unstable.
Backporting with JDK-8035395 makes JMXStartStopTest.java more stable.

Thank you.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8035395](https://bugs.openjdk.org/browse/JDK-8035395) needs maintainer approval

### Issue
 * [JDK-8035395](https://bugs.openjdk.org/browse/JDK-8035395): sun/management/jmxremote/startstop/JMXStartStopTest.java fails intermittently: Port already in use (**Bug** - P4 - Approved)

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/429/head:pull/429` \
`$ git checkout pull/429`

Update a local copy of the PR: \
`$ git checkout pull/429` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/429/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 429`

View PR using the GUI difftool: \
`$ git pr show -t 429`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/429.diff">https://git.openjdk.org/jdk8u-dev/pull/429.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/429#issuecomment-1909524402)

### CODE REVIEW

**Review by phohensee — APPROVED** [review]

**Comment by phohensee** [comment]
windows-x86 GHA builds seem broken due to unrelated issue.

**Comment by ktakakuri** [comment]
/approval request I would like to backport this issue to 8u because  the test fails also with jdk8.
There is no risk because it is a test-only fix.
Only with this fix , the test will not passed.
This backport is followed by a JDK-8066708 backport to stabilize the test.

**Review by gnu-andrew — APPROVED** [review]
Backport looks good to me.

I see some differences with the 11u version, but this is because 8u got JDK-8044135 first, whereas this comes after this patch in 11u. JDK-8044135 removes `test_13` and `test_14` which were patched by 8035395 in 11u, and also alters the method signatures.

File: jdk/test/sun/management/jmxremote/startstop/JMXStartStopTest.java [hunk 1]
```diff
@@ -702,29 +749,30 @@ static void test_10() throws Exception {
...
-            testNoConnect(port1);
+            testNoConnect(pa.getPort1());
 
             jcmd(CMD_STOP);
             jcmd(CMD_START,
                 "jmxremote.ssl=false",
-                "jmxremote.port=" + port1
+                "jmxremote.port=" + pa.getPort1()
             );
             testConnect(port1);
```
Comment (mrserb) [inline jdk/test/sun/management/jmxremote/startstop/JMXStartStopTest.java:768]: fyi: change to port1 is missed, see https://github.com/openjdk/jdk8u-dev/pull/429

**Pull request closed without merging**